compiler/rustc_mir_transform/src/match_branches.rs

```diff
@@ -263,33 +264,59 @@ impl<'tcx> SimplifyMatch<'tcx> for SimplifyToIf {
     }
 }
 
+/// Check if the cast constant using `IntToInt` is equal to the target constant.
+fn can_cast(
+    from_val: impl Into<u128>,
+    from_size: Size,
+    target_scalar: ScalarInt,
+    from_is_signed: bool,
+) -> bool {
+    let from_val = from_val.into();
+    let from_scalar = ScalarInt::try_from_uint(from_val, from_size).unwrap();
+    let to_size = target_scalar.size();
+    let cast_scalar = match to_size.cmp(&from_size) {
+        // Truncate to the target size.
+        Ordering::Less => ScalarInt::truncate_from_uint(from_val, to_size).0,
+        Ordering::Equal => from_scalar,
+        Ordering::Greater => {
+            if from_is_signed {
+                // Extend to the target size using sext.
+                ScalarInt::try_from_int(from_scalar.to_int(from_size), to_size).unwrap()
+            } else {
+                // Extend to the target size using zext.
+                ScalarInt::try_from_uint(from_scalar.to_uint(from_size), to_size).unwrap()
+            }
+        }
+    };
+    cast_scalar == target_scalar
+}
```

Comment:
This code does not exactly match what the interpreter does in `cast_from_int_like`. Should we reuse similar code?

Reply:
Updated.
When I read this part of the code earlier, I tried to call it directly, which seemed a bit difficult. I copied part of the code. Maybe I should put such code in `rustc_middle`?

Reply:
`rustc_const_eval` is accessible from this crate, so putting is there is enough.

Reply:
Updated, I have squashed them into two commits.